File: 1-twoSum/Cpp/twoSum.cpp
#include <unordered_map>
#include <vector>
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {        
        std::unordered_map<int, int> smap{};
        auto nsize = nums.size();
        for(auto i = 0; i < nsize; i++){
            auto x = target - nums[i];
            if(smap.find(x) !=smap.end()){
                return {smap[x], i};
            }
            smap.insert({nums[i], i});
        }
        return {0,0};
    }
};

File: 1-twoSum/JS/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let seen_list = new Map();
  let n_len = nums.length;
  for (let i = 0; i < n_len; i++) {
    let x = target - nums[i];
    if (seen_list.has(x)) {
      return [seen_list.get(x), i];
    }
    seen_list.set(nums[i], i);
  }
};

module.exports = twoSum;
